internal/command/webbrowser/exec_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package webbrowser
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+)
+
+const fakeBrowserLaunchCmdOutputEnvName = "OPENTOFU_WEBBROWSER_EXEC_TEST_OUTPUT"
+
+// TestMain overrides the test entrypoint so that we can reuse the test
+// executable as a fake browser launcher command when testing
+// [NewExecLauncher].
+func TestMain(m *testing.M) {
+	if f := os.Getenv(fakeBrowserLaunchCmdOutputEnvName); f != "" {
+		err := fakeBrowserLauncherCommand(f)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "fake browser launcher failed: %s", err)
+			os.Exit(1)
+		}
+		os.Exit(0)
+	}
+	os.Exit(m.Run())
+}
+
+func fakeBrowserLauncherCommand(outputFilename string) error {
+	// The "exec" browser launcher must pass the URL to open in the
+	// first argument to the executable it launches.
+	url := os.Args[1]
+	return os.WriteFile(outputFilename, []byte(url), os.ModePerm)
+}
+
+func TestExecLauncher(t *testing.T) {
+	// For this test we re-use the text executable as a fake browser-launching
+	// program, through the special logic in [TestMain].
+	fakeExec := os.Args[0]
+
+	tempDir := t.TempDir()
+	outputFile := filepath.Join(tempDir, "browser-exec-launcher-test")
+	t.Setenv(fakeBrowserLaunchCmdOutputEnvName, outputFile)
+
+	launcher := NewExecLauncher(fakeExec)
+	err := launcher.OpenURL("http://example.com/")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	result, err := os.ReadFile(outputFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if got, want := string(result), "http://example.com/"; got != want {
+		t.Errorf("wrong URL written to output file\ngot:  %s\nwant: %s", got, want)
+	}
+}
+
+func TestParseBrowserEnv_success(t *testing.T) {
+	// ParseBrowserEnv only actually needs to work on Unix-like systems, so
+	// the set scenario below is not written to be portable.
+	// There's no runtime equivalent of the "unix" build tag, so we just
+	// explicitly test the two main Unix OSes we support here.
+	if runtime.GOOS != "linux" && runtime.GOOS != "darwin" {
+		t.Skip("ParseBrowserEnv is only for unix systems")
+	}
+
+	tmpDir := t.TempDir()
+	fakeExec := filepath.Join(tmpDir, "fake-launch-browser")
+	err := os.WriteFile(fakeExec, []byte(`not a real program`), 0755)
+	if err != nil {
+		// NOTE: This test requires the temp directory to be somewhere that
+		// allows executables, so this won't work if the temp directory is
+		// on a "noexec" mount on a Unix-style system.
+		t.Fatal(err)
+	}
+	// Temporarily we'll reset the search path to just our temp directory
+	os.Setenv("PATH", tmpDir)
```

Comment:
```suggestion
	t.Setenv("PATH", tmpDir)
```
`t.Setenv` should be used instead since the only difference is that the suggested one reverts the change on test cleanup.

Reply:
You're right... I don't know why I wrote `os` here since I was intending to use `t` as you suggested. I guess it was just habit. I will fix it.